src/runtime.c

```diff
@@ -536,7 +528,15 @@ int main(int argc, char *argv[]) {
      * functionality specifically for builds used by appimaged.
      */
     if (getenv("TARGET_APPIMAGE") == NULL) {
-        strcpy(appimage_path, "/proc/self/exe");
+        // for some reason, `fopen("/proc/self/exe", "rb")` tries to open
+        // the dynamic linker when running under gcompat, even though `readlink()`
+        // gives the right result, so use `readlink()` here
+        ssize_t len = readlink("/proc/self/exe", appimage_path, sizeof(appimage_path));
```

Comment:
Consider using `realpath`. `readlink` should work functionally, but we could also just use absolute, fully resolved paths. Then, the name `full_path` would also make more sense.

Reply:
That could break some existing portable installations:  `realpath` resolves all layers of symlinks, whereas `readlink` resolves just the one layer given to it.  For example:

* `/tmp/a.AppImage` points to `/home/user/b.AppImage`.
* `/home/user/b.AppImage` points to `Downloads/c.AppImage`.
* Currently, the portable home/config directories would be created at `/home/user/b.AppImage.{home,config}`, since that's what `readlink` would return.
* Using `realpath`, they would instead be created at `/home/user/Downloads/c.AppImage.{home,config}`, since all layers of symlinks would be resolved.
* The directories used when _running_ the AppImage would also change, since the result of `readlink` is also used there.

Regardless, since my only goal with this PR is to get AppImages running in Alpine/gcompat environments, I didn't want to change how symlinks are resolved except for resolving `/proc/self/exe` early.

I also noticed that `realpath` _is_ used when `$TARGET_APPIMAGE` is set.  Again, I tried to preserve the existing behavior there (but making the help and portable options consistent with what value is actually used later).

Reply:
It could also break AppImages (with or without portable directories) which use `$APPIMAGE`, since `appimage_full_path` is used for that, and _without_ `$TARGET_APPIMAGE`, that is the result of `readlink()` both before and after this PR.

Reply:
I see, in this code path `readlink` makes more sense.